resources/windows/wix/customactions.wxs.erb

```diff
@@ -259,22 +328,36 @@ Sub ExecuteCommand (Command)
   End If
 
   fso.DeleteFile(tempFilePath)
-End Sub
+End Function
 
-Sub ResetPermissionsIfExists (Path, GrantAdmins)
+Function ResetPermissionsIfNotMatches (PathToCheck, PathToReset, SDDL, GrantAdmins)
 
-  If (fso.FolderExists(Path)) Then
-    Log "Modifying found directory : " & Path, False
+  If (fso.FolderExists(PathToCheck) Or fso.FileExists(PathToCheck)) Then
+    Log "Inspecting found path : " & PathToCheck, False
 
-    If GrantAdmins = True Then
-      ExecuteCommand comspec & " /C " & icacls & "  """ & Path & "\*"" /grant ""*S-1-5-32-544:(F)"" ""*S-1-5-18:(F)"" /T /C"
-    End If
+    Dim checkSDDL : checkSDDL = GetSDDL(PathToCheck)
+    If checkSDDL <> SDDL Then
+      Log "Resetting " & PathToReset & " based on unexpected SDDL for " & PathToCheck & " with " & checkSDDL & " expected to be " & SDDL, False
+      If GrantAdmins = True Then
+        ' while it is possible to call takeown directly, we avoid a WshShell stream problem with stderr this way
+        ' https://machine-cycle.blogspot.com/2008/06/pipe-dreams-or-vbscript-spawned.html
+        Log ExecuteCommand(comspec & " /C " & takeown & " /F """ & PathToReset & "\*"" /R /A /D N"), False
```

Comment:
FYI - by running `takeown PuppetLabs\puppet\*`, ownership is reset on all children, but not `puppet` itself. The end result is that `puppet` has an owner and group of `SYSTEM`, while all subdirectories are owner of `Administrators` and group of `SYSTEM`